### PR TITLE
feat(wakatime): add segment

### PR DESCRIPTION
**Pull request by JanDeDobbeleer**
### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes/features)

### Description

Different approach, leverages temple functions (which can be reused by the `executiontime` segment in templates)

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

**Comment by JanDeDobbeleer** [comment]
@Khaos66 this has a few advantages:

- no longer using the complex settings (just add the URL with your params)
- leverage a template function to format the Seconds

**Comment by Khaos66** [comment]
@JanDeDobbeleer I get your point now ;D
Yes that's simpler on this side. Though, the user has to read the API docs and build the URL on its own now.
But we're serving devs here, so this should be ok

File: docs/docs/segment-wakatime.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        "cache_timeout": 10,
+        "http_timeout": 500
+    }
+},
+```
+
+## Properties
+
+- url: `string` - Your Wakatime summaries URL, inclding the API key. Example above. You'll know this works if you can
+- curl it yourself and get a single value. - defaults to ``
```
Comment (Khaos66) [inline docs/docs/segment-wakatime.md]: You could add a link to the actual API doc: 
https://wakatime.com/developers#summaries

**Comment by Khaos66** [comment]
@JanDeDobbeleer This segment is kind of generic now. Apart from the response structs there is not much left, that is very wakatime specific...
Would it be possible to remove these structs as well? Then this would be a generic `webapi` segment and could be used for ANY rest api =)
I'm not sure how to document wakatime then. Because I would want to point users on that great service ;D 
Maybe a sub-section in the docs where all kinds of webapis are documented, so the user can copy&paste an api easily?
What do you think?

**Comment by JanDeDobbeleer** [comment]
@Khaos66 there's a difference between architecture for oh-my-posh and how we expose that to the outside. I would wait to streamline this once we have generics as the parsing of the json is going to be cumbersome otherwise. But you're right, we can reuse this logic.

**Pull request merged**